cmd/data-aggregation-api/main.go

```diff
@@ -40,6 +40,20 @@ var (
 	builtBy = "unknown"
 )
 
+func dispatchSingleRequest(incoming chan struct{}) chan bool {
+	outgoing := make(chan bool)
+
```

Comment:
I think you could use a single channel with a size of 1 (since they're bidirectional).

Once you have sent the notification, you could wait until the receiver has sent a notification back to you to signal the build is finished ?

Reply:
It would be blocking.
If there is an ongoing build, the user who does the request would have to wait the end of the build to get the response.
It means the user would have to wait minutes before getting the response back.

Reply:
note: I've just updated the definition to make `incoming` channel read-only. (which is a detail)

Reply:
plus, if you use the same channel in bidirectional, how to know if the build is responding or if it is just another user calling the same endpoint in the meantime?

Reply:
also, I have changed the channel from bool to struct, to make it more explicit I am handling the close channel in job.go:196